src/main/java/com/crud/tasks/controller/TaskController.java

```diff
@@ -32,6 +33,7 @@ public ResponseEntity<TaskDto> getTask(@PathVariable Long taskId) throws TaskNot
 
     @DeleteMapping(value = "{taskId}")
     public ResponseEntity<Void> deleteTask(@PathVariable Long taskId) {
+        service.deleteTask(taskId);
```

Comment:
Long czy long? I dlaczego?

Reply:
Long - wrapper, ponieważ w kolekcjach i interfejsach operujemy na typach obiektowych (generic types)?

Reply:
Ale tu nie masz kolekcji - masz parametr metody

Reply:
Zasięgnąłem porad mądrych osób i Long, ponieważ w klasie Task id również jest typu Long, przez co nie musimy rzutować typów przy odwołaniach do repository

Reply:
To ja sie z tymi madrymi nie zgodze :) generalnie typ w encji tez powinien byc long a nie Long :) Long pozwala na to, by wartosc byla nullem (mamy obiekt). Czy chcesz kasowac zadanie z id=null? Rzutowanie akurat tu jest boxingiem, a ta operacja jest prawie darmowa

Reply:
Właśnie o zapobieganie usuwania id=null miałem pytać w środę bo mi nie wychodziło wrzucenie do tej metody wyjątku TaskNotFoundException. Czyli tutaj zmienić typ na long czy coś jeszcze jest do poprawy?

Reply:
Zmienic na long

Reply:
Teraz jeszcze wykombinowałem coś takiego aby uodpornić metodę na usuwanie id=null, przetestowałem w Postmanie i działa:
`    @DeleteMapping(value = "{taskId}")
    public void deleteTask(@PathVariable long taskId) throws TaskNotFoundException {
        if (service.getTask(taskId) != null) {
            service.deleteTask(taskId);
        } else {
            throw new TaskNotFoundException();
        }
    }`